hack/verify-lint.sh

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env bash
+
+# standard bash error handling
+set -o nounset # treat unset variables as an error and exit immediately.
+set -o errexit # exit immediately when a command fails.
+set -E         # needs to be set if we want the ERR trap
+
+readonly CURRENT_DIR=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+readonly ROOT_PATH="${1:-$( cd "${CURRENT_DIR}/.." && pwd )}" # first argument or root of the project
+readonly TMP_DIR=$(mktemp -d)
+readonly GOLANGCI_LINT_VERSION="v1.38.0"
+
+source "${CURRENT_DIR}/utilities.sh" || { echo 'Cannot load CI utilities.'; exit 1; }
+
+cleanup() {
+    rm -rf "${TMP_DIR}" || true
+}
+
+trap cleanup EXIT SIGINT
+
+golangci::install() {
```

Comment:
Do we need to install the tool?
In my opinion it should be in buildpack. 
In case of developer machine, we should add this tool to requirements and add description how to install it or put it on other script

Reply:
This script is built so that it works on both CI and developer machines.
But there is also a flag that can be passed to skip the installation of `golangci lint`, so if the environment already has it, the flag can be passed and it will not be installed.

Reply:
By default I would turn off installing.

Reply:
What do u think to turn it off by default?
Because other tools are also not installed every time.

Reply:
In hydroform we are calling the `before-commit.sh` script with the `ci` flag. I will change it so that when using that flag `golangci-lint` is not installed per default.

Reply:
Added linter to the `golang-toolbox` image: https://github.com/kyma-project/test-infra/pull/3441#event-4495789770
Updated prow jobs using linter: https://github.com/kyma-project/test-infra/pull/3447

Disabled installation of `golangci-lint` on CI flagged executions. You can see on the pipeline that the linter is not being downloaded anymore.

Reply:
Several reason why the script shouldn't download the tool by default:
- it makes this script slower for developers. 
- In case that the download page will be unavailable and we have this tool locally, the developer will stuck because the linter target want to install something.
- It's violation of single responsibility principle, because this downloads and validate, but the name is `lint-something`
- this project is written in Go, why we don't always download Go when compiling it?

If you want to download something, you can put it in seperate target and in `linter` target. 
It makes more sense to check if the tool is available. When there is no such tool print error and solution how to resolve it.

Reply:
Installing linter is also one time action.

Reply:
We can add the check to see if the tool is installed yes, still using an older version of the linter might have different results locally and on the pipeline. And that will be very confusing.

What we can do is printing out the installation command with instructions in case the binary is not found or the version is not the expected one.

Reply:
Hi @dbadura,

I have changed the script so that it now verifies the golangci-lint installation and prints install/update instructions if not correct.
Now the developer can install once himself the linter or update it if we increase the minimum version.

Reply:
Looks much better now ;)